### PR TITLE
Update manylinux build scripts

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

Update manylinux build scripts. 

**Motivation and Context**
- Why is this change required? What problem does it solve?

Fix pcre download URL. The old one is broken.

- If it fixes an open issue, please link to the issue here.

Get https://github.com/pypa/manylinux/pull/1218 into our repo.

### CODE REVIEW

File: cgmanifests/generated/cgmanifest.json [hunk 1]
```diff
@@ -17,8 +17,8 @@
         "Type": "other",
         "other": {
           "Name": "automake",
-          "Version": "1.16.4",
-          "DownloadUrl": "http://ftp.gnu.org/gnu/automake/automake-1.16.4.tar.gz"
+          "Version": "1.16.5",
```
Comment (yuslepukhin) [inline cgmanifests/generated/cgmanifest.json:20]: >Version": "1.16.5", [](http://example.com/codeflow?start=11&length=19)

just curious,how you generated these changes?

File: tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_cuda11 [hunk 2]
```diff
@@ -30,12 +30,19 @@ ENV PKG_CONFIG_PATH=/usr/local/lib/pkgconfig
 COPY build_scripts/fixup-mirrors.sh /usr/local/sbin/fixup-mirrors
 
 # setup entrypoint, this will wrap commands with `linux32` with i686 images
-COPY build_scripts/install-entrypoint.sh build_scripts/update-system-packages.sh /build_scripts/
-RUN bash /build_scripts/install-entrypoint.sh && rm -rf /build_scripts
+COPY build_scripts/install-entrypoint.sh \
+     build_scripts/update-system-packages.sh \
+     build_scripts/build_utils.sh \
```
Comment (yuslepukhin) [inline tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_cuda11:35]: >build_scripts/build_utils.sh  [](http://example.com/codeflow?start=5&length=29)

This seems to be a new script, but I am not seeing it among the added.

File: cgmanifests/generated/cgmanifest.json (hunk 1 above)
Comment (snnn) [reply cgmanifests/generated/cgmanifest.json:20]: 3 ways diff and merge. 

1. The files in our repo
2. The files in the upstream repo with the specific commit id which can be found in the README.txt in our build_script dir.
3. The files in the upstream repo with the latest version

**Review by yuslepukhin — APPROVED** [review]
:shipit:

**Pull request merged**